### PR TITLE
Add ability to specify custom keyset reader/writer for AndroidKeysetManager

### LINKED ISSUE #264: [android] Add ability to specify custom keyset reader/writer for AndroidKeysetManager
I would like to use custom storage for keysets instead of SharedPreference, but still use AndroidKeysetManager for easy handling master key with Android keychain.

### PR DESCRIPTION
**Pull request by varahash**
Fixes #264

### CODE REVIEW

**Comment by varahash** [comment]
@googlebot I signed it!

File: java/src/main/java/com/google/crypto/tink/integration/android/AndroidKeysetManager.java [hunk 1]
```diff
@@ -100,13 +103,13 @@ private AndroidKeysetManager(Builder builder) throws GeneralSecurityException, I
     reader = builder.reader;
     if (reader == null) {
       throw new IllegalArgumentException(
-          "need to specify where to read the keyset from with Builder#withSharedPref");
+          "need to specify where to read the keyset from with Builder#withSharedPref or Builder#withKeysetReaderWriter");
```
Comment (chuckx) [inline java/src/main/java/com/google/crypto/tink/integration/android/AndroidKeysetManager.java]: Small nit, but can you modify this (and the instance below) to adhere to https://google.github.io/styleguide/javaguide.html#s4.4-column-limit

Essentially:
```suggestion
          "need to specify where to read the keyset from with Builder#withSharedPref or "
              + "Builder#withKeysetReaderWriter");
```

Comment (varahash) [reply java/src/main/java/com/google/crypto/tink/integration/android/AndroidKeysetManager.java]: Done.

File: java/src/main/java/com/google/crypto/tink/integration/android/AndroidKeysetManager.java [hunk 2]
```diff
@@ -148,6 +151,13 @@ public Builder withSharedPref(Context context, String keysetName, String prefFil
       return this;
     }
 
+    /** Reads and writes the keyset from provided reader/writer. */
+    public Builder withKeysetReaderWriter(KeysetReader reader, KeysetWriter writer) {
```
Comment (thaidn) [inline java/src/main/java/com/google/crypto/tink/integration/android/AndroidKeysetManager.java]: I think this should be split off to withKeysetReader and withKeysetWriter.

**Comment by juergw** [comment]
We decided that we don't want to add this to the AndroidKeysetManager. Instead of using custom readers/writers, it is preferable that users implements the reading/writing themselves.

**Pull request closed without merging**